cmd/gateway-gcs.go

```diff
@@ -908,29 +910,40 @@ func (l *gcsGateway) CompleteMultipartUpload(bucket string, key string, uploadID
 	if err != nil {
 		return ObjectInfo{}, gcsToObjectError(traceError(err), bucket, key)
 	}
+
+	if len(uploadedParts) > maxPartCount {
```

Comment:
If we have input list > 1024 we can't concatenate anyways.. not sure if i understand you correct.